db/fetcher/encoded_doc.go

```diff
@@ -41,90 +54,104 @@ func (e encProperty) Decode() (client.CType, any, error) {
 		return ctype, nil, err
 	}
 
+	decodedVal, err := Decode(e.Desc, val)
+	return ctype, decodedVal, err
+}
+
+func Decode(fieldDesc client.FieldDescription, val any) (any, error) {
```

Comment:
I thought I had a todo/code- comment somewhere reminding me to look at doing just that. I'll add a PR todo too just in case that is lost.

- [ ] Move Decode?